### DISCUSSION TITLE
0528 피어세션일지

### DISCUSSION THREAD

**Discussion by arabae** [bcaitech1/p4-dkt-freshtomato#43] [General]
**이유경멘토님**
DST멘토, 석박통합과정, 현재는 박사과정
language 데이터 사이에서 anomaly detection을 현재 연구하고있음

**프로젝트**
- 한국어 문서요약 데이터셋 구축하고 모델 개발 및 배포예정
- 산업경영공학과 → 제조업에서 수요가 있어서 실제 제조업데이터를 어떻게 DL/ML 할 것인지 (좋은 퀄리티의 생산이 가능하도록)
- information extraction에서 정보 이용하여 주가를 예측

DKT를 전문X — domain knowledge가 부족한 상태니까.. 수업을 들으려고 하고있음.. (감동..😭)
- 이전에 멘토링했을 때는 DST 관련도 했지만 어떻게 논문을 읽고, 정리하는지 협업을 어떻게 진행하는지에 대해 이야기함

(재훈)
주가예측, 공정기술에서 접해볼 수 없는 현업에서 다양한 프로젝트에 어떻게 접근해야할지 궁금
- DKT 멘토링도 좋지만, 경험하신 부분에서 멘토링도 좋을 것 같음
- 실제 real 데이터를 만났을 때 어떻게 처리하고, 진행했는지 정리해서 얘기해주실 예정

---

매주마다 설문지를 줄건데.. 익명으로 이런 부분을 해주셨으면 좋겠다 (2일전에 취합)
- 그 주마다 생기는 이슈를 잡을 것
- DKT에 deep하게 잡을 것인지, 다양한 것을 다룰지 방향을 정했으면 좋겠음

2주차까지는 deep하게 뜯어보는게 좋을 것 같고, 3주차때부터 앙상블을 어떻게할건지 생각해보는게 좋을 것 같음
- PLM 등에서 궁금하다면 물어보면 많이 알려드리겠습니다.
- DKT 멘토님께 연락해서라도 꼭 알려드릴테니 많이 질문해주세요..

feature를 고려하는게 가장 중요하다고 느껴짐
- 어떤 feature를 사용했는지 치열하게 토론하는게 좋을 것 같음
→ 나중에 고도화될수록 이런 중요성이 낮을 수가 있음

같은 맡은 role를 명확하게 하는게 좋을 것 같음!!
- DKT에서 어떻게 앙상블하는지 찾아봐야할 것 같은데
- 데이터셋을 잘 뜯어보면 좋을 것 같음
- 이 task의 목적은 무엇인지 목적을 달성하기 위해, 어떤 feature가 사용되는지 확인

(재훈) 검증전략을 어떻게 세울 것인가?
- valid랑 LB랑 유사하게 가져가야한다고 생각
- stratified 기준을 여러개 사용, 여러 가지 seed → AUC metric에서 LB와 valid가 차이가 많았음
(멘토님) 이번에 private set이 있는지?
- LB와 차이가 있다는 것은 분포가 다르다는 이야기
    - 제출한 파일이랑 LB의 점수를 같이 보면 그 경향성을 알 수 있을 것
    - 모델이 어떤걸 잘하고, 못하는지 파악하는게 중요
    - 각 데이터에 대해 못 맞추면 이러한 feature가 필요할 것 같음
    - LB에 제출해보면서 이번주내에 확인하는게 중요한 것 같음
- **output에 대한 분석이 아주 중요!!**
    - 각각 찾아서 피어세션에서 공유하는게 좋을 것 같음
    - 나눠서 해보고, 같이 이야기
- 질문을 엄청 많이 하는게 중요

---

**valid랑 LB 비교 (output의 경향 비교)**
▶ 적어도 자기가 냈던 파일에 대해서 분석해오기

(민용)
- validation이랑 LB score가 너무 다르다면 분포가 다르니까, test 분포랑 비슷하게 맞춰줘야하는데
- 이를 맞춰주기 위해서는 EDA가 중요하고, 최대한 valid랑 test 분포를 비슷하게 맞춰주고 여기서 시도하기 위해서 여러가지 제출을 해볼텐데 여기서 낮은지 높은지를 파악
- 관계를 파악하는 과정에서 feature를 추가할 수 있을 것 같다고 이해
- 지금 validation이랑 test 데이터 셋의 분포를 비슷하게 맞춰줘야할지 어려운 것 같음

(유라)
- 각자 feature를 뽑아보라는 얘기가 기억에 남고..
- 모델이 어떤거에 취약한지 분석
- 엔지니어링한 feature에 따라 정답 분포를 확인하고 어떻게 취약한지 확인

(재훈)
- stratified를 기준으로 삼았던 기준이 있는데 이게 유의한지 아닌지 판단하는게 가능할 것 같다고 얘기해줬는데
- 어떻게 유의한지 판단? — LB와 valid가 비슷해질 때를 의미하는건지

(민용)
- 여기서 만든 기준을 우리의 입력 데이터로 만드는 feature로 연결될지는 잘 모르겠음..

(재훈)
추가했던 feature가 유의미한지 파악하는건 LB 점수를 보고 아는건가?
- (민용) 그러면 더더욱 EDA를 해야한다..
- 사람을 기준으로 나누고 있음 — 테스트 사람의 분포와 비슷하게 valid를 맞춰줘야하는데.. train과 valid를 나눌 때 사람을 기준으로 나누면서 테스트에 속해있는 특성의 분포가 비슷해야될 것 같음..
- 이거를 파악하기 위해 정답률, 얼마나 잘 맞췄는지, 0/1 나누고, 어떤 KT를 나눴는지도 해봤지만.. 이게 좋은지 파악을 못하고 있는 것 같음

---

➕ **_실험 변경 사항_**
K-fold로 고정하고, seed는 42 사용

**Comment by gvsteve24** [comment]
완벽해!!! 고마워 아라야 😃
